atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/CellTest.java

```diff
@@ -0,0 +1,86 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.logsafe.exceptions.SafeNullPointerException;
+import java.nio.charset.StandardCharsets;
+import org.junit.Test;
+
+public final class CellTest {
+
+    @Test
+    public void create() {
+        Cell cell = Cell.create(bytes("row"), bytes("col"));
+        assertThat(cell.getRowName()).isEqualTo(bytes("row"));
+        assertThat(cell.getColumnName()).isEqualTo(bytes("col"));
+        assertThatThrownBy(() -> Cell.create(null, bytes("col"))).isInstanceOf(SafeNullPointerException.class);
+        assertThatThrownBy(() -> Cell.create(bytes("row"), null)).isInstanceOf(SafeNullPointerException.class);
+        assertThatThrownBy(() -> Cell.create(bytes(""), bytes(""))).isInstanceOf(SafeIllegalArgumentException.class);
+        assertThatThrownBy(() -> Cell.create(bytes("row"), bytes(""))).isInstanceOf(SafeIllegalArgumentException.class);
+        assertThatThrownBy(() -> Cell.create(bytes(""), bytes("col"))).isInstanceOf(SafeIllegalArgumentException.class);
+        assertThatThrownBy(() -> Cell.create(bytes("row"), bytes("x".repeat(Cell.MAX_NAME_LENGTH + 1))))
+                .isInstanceOf(SafeIllegalArgumentException.class);
+        assertThatThrownBy(() -> Cell.create(bytes("x".repeat(Cell.MAX_NAME_LENGTH + 1)), bytes("col")))
+                .isInstanceOf(SafeIllegalArgumentException.class);
+    }
+
+    @Test
+    @SuppressWarnings("ConstantConditions") // explicitly testing conditions
+    public void isNameValid() {
+        assertThat(Cell.isNameValid(bytes("row"))).isTrue();
+        assertThat(Cell.isNameValid(null)).isFalse();
+        assertThat(Cell.isNameValid(new byte[0])).isFalse();
+        assertThat(Cell.isNameValid(bytes("x"))).isTrue();
+        assertThat(Cell.isNameValid(bytes("x".repeat(Cell.MAX_NAME_LENGTH + 1))))
+                .isFalse();
+    }
+
+    @Test
+    public void compareTo() {
+        assertThat(Cell.create(bytes("row"), bytes("col")))
+                .isEqualByComparingTo(Cell.create(bytes("row"), bytes("col")));
+        assertThat(Cell.create(bytes("row"), bytes("col")))
+                .isNotEqualByComparingTo(Cell.create(bytes("row2"), bytes("col")));
+        assertThat(Cell.create(bytes("row"), bytes("col")))
+                .isNotEqualByComparingTo(Cell.create(bytes("row2"), bytes("col2")));
+        assertThat(Cell.create(bytes("row1"), bytes("col"))).isLessThan(Cell.create(bytes("row2"), bytes("col")));
+        assertThat(Cell.create(bytes("row1"), bytes("col"))).isGreaterThan(Cell.create(bytes("row0"), bytes("col")));
```

Comment:
Also, the `compareTo` method has two branches, and this only tests the first (where the row differs); perhaps worth testing the compareTo part when row is the same and col is less or greater?